master/buildbot/reporters/http.py

```diff
@@ -97,17 +97,28 @@ def send(self, build):
 class HttpStatusPush(HttpStatusPushBase):
     name = "HttpStatusPush"
 
-    def checkConfig(self, serverUrl, user, password, **kwargs):
+    def checkConfig(self, serverUrl, user=None, password=None, auth=None, format_fn=None, **kwargs):
+        if user is not None and auth is not None:
+            config.error("Only one of user/password or auth must be given")
+        if user is not None:
+            config.warnDeprecated("0.9.1", "user/password is deprecated, use 'auth=(user, password)'")
+        if (format_fn is not None) and not callable(format_fn):
+            config.error("format_fn must be a function")
         HttpStatusPushBase.checkConfig(self, **kwargs)
 
-    def reconfigService(self, serverUrl, user, password, **kwargs):
+    def reconfigService(self, serverUrl, user=None, password=None, auth=None, format_fn=None, **kwargs):
         HttpStatusPushBase.reconfigService(self, **kwargs)
         self.serverUrl = serverUrl
-        self.auth = (user, password)
+        self.auth = auth
+        if user is not None:
+            self.auth = (user, password)
```

Comment:
One can argue about the "cleaness" of both. The version I wrote made it clear that ``user is not None`` is a special case, and the rest the normal case. In your version, ``user is not None`` is the very first check, and the rest is a fallback, which convey a slightly different meaning. Additionally, from python zen: _Flat is better than nested._.

I'll fix it anyway.